thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -80,6 +77,24 @@ class CIndexReadSlaveBase : public CSlaveActivity
     rowcount_t rowLimit = RCMAX;
     bool useRemoteStreaming = false;
 
+    template<class StatProvider>
+    class CCaptureIndexStats
+    {
+        CRuntimeStatisticCollection &stats;
+        StatProvider &statProvider;
+        unsigned __int64 startSeeks = 0, startScans = 0;
+    public:
+        inline CCaptureIndexStats(CRuntimeStatisticCollection &_stats, StatProvider &_statProvider) : stats(_stats), statProvider(_statProvider)
+        {
+            startSeeks = statProvider.querySeeks();
+            startScans = statProvider.queryScans();
+        }
+        inline ~CCaptureIndexStats()
+        {
+            stats.mergeStatistic(StNumIndexSeeks, statProvider.querySeeks() - startSeeks);
```

Comment:
Slightly concerned about the extra overhead of this call.  Might be worth considering a fastpath for this particular case of summing a stat.